src/ZoneServer/Skills/Handlers/Archer/Wugushi/Zhendu.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using Melia.Shared.Game.Const;
+using Melia.Shared.L10N;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Pads;
+
+namespace Melia.Zone.Skills.Handlers.Archer.Wugushi
+{
+	/// <summary>
+	/// Handler for the Wugushi skill Zhendu.
+	/// </summary>
+	[SkillHandler(SkillId.Wugushi_Zhendu)]
+	public class Zhendu : ISelfSkillHandler
+	{
+		/// <summary>
+		/// Handles skill, applying a buff to the caster.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="dir"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Direction dir)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			if (caster.IsAbilityActive(AbilityId.Wugushi7))
+			{
+				var spendSp = skill.Properties.GetFloat(PropertyName.SpendSP) / 2;
+				if (!caster.TrySpendSp(spendSp))
```

Comment:
By calling this twice, you're actually spending SP twice. Is this intentional?

Reply:
![image](https://github.com/user-attachments/assets/eabe91b8-2b00-4b0f-b8d5-3f7564cf0609)

Yes, SP increased by 50%.

Reply:
Ability.ies has columns called SkillCategory and AddSpend that specify which abilities increase the SP costs of which skills.  Maybe this could be added to the DB and applied automatically?

Reply:
We can certainly look into that, though I'm not fond of how `AddSpend` can also affect the cooldown and potentially other values they might add later. They're really more script than data, modifying values if certain conditions are met.